packages/react-link/src/components/Link/useLinkState.ts

```diff
@@ -1,62 +1,28 @@
 import * as React from 'react';
-import { Enter, Space } from '@fluentui/keyboard-keys';
+import { getCode, EnterKey, SpacebarKey } from '@fluentui/keyboard-key';
 import type { LinkState } from './Link.types';
 
 /**
- * The useLink hook processes the Link draft state.
+ * The useLink hook processes the Link draft state.root.
  * @param state - Link draft state to mutate.
  */
 export const useLinkState = (state: LinkState): LinkState => {
-  const { as, disabled, disabledFocusable, href, onClick, onKeyDown: onKeyDownCallback, type } = state;
+  const { disabled, disabledFocusable } = state;
+  const { onClick, onKeyDown, role, type } = state.root;
 
-  const onNonAnchorOrButtonKeyDown = (ev: React.KeyboardEvent<HTMLAnchorElement | HTMLButtonElement | HTMLElement>) => {
-    onKeyDownCallback?.(ev);
-
-    const key = ev.key;
-    if (!ev.defaultPrevented && onClick && (key === Enter || key === Space)) {
-      // Translate the keydown enter/space to a click.
-      ev.preventDefault();
-      ev.stopPropagation();
-
-      onClick((ev as unknown) as React.MouseEvent<HTMLAnchorElement | HTMLButtonElement | HTMLElement>);
-    }
-  };
-
-  // Adjust props depending on the root type.
-  if (typeof as === 'string') {
-    if (as === 'a') {
-      state.href = disabled ? undefined : href;
-      state.tabIndex = disabled && !disabledFocusable ? undefined : 0;
-    }
-    // Remove the href, rel and target props for all non-anchor elements.
-    else {
-      state.href = undefined;
-      state.rel = undefined;
-      state.target = undefined;
-
-      // Add 'role=link' for all non-anchor elements.
-      state.role = 'link';
-
-      // Add the type="button" prop for button elements.
-      if (as === 'button') {
-        state.type = type ? type : 'button';
-      }
-      // Add keydown event handler and 'tabIndex=0' for all other elements.
-      else {
-        state.onKeyDown = onNonAnchorOrButtonKeyDown;
-        state.tabIndex = disabled && !disabledFocusable ? undefined : 0;
-      }
-    }
+  // Add href and tabIndex=0 for anchor elements.
+  if (state.root.as === 'a') {
+    state.root.href = disabled ? undefined : state.root.href;
+    state.root.tabIndex = disabled && !disabledFocusable ? undefined : 0;
   }
-  // Add keydown event handler, 'role=link' and 'tabIndex=0' for all other elements.
+  // Add 'role=link' and 'type=button' for button elements.
   else {
-    state.onKeyDown = onNonAnchorOrButtonKeyDown;
-    state.role = 'link';
-    state.tabIndex = disabled && !disabledFocusable ? undefined : 0;
+    state.root.role = role || 'link';
+    state.root.type = type || 'button';
   }
 
   // Disallow click event when component is disabled and eat events when disabledFocusable is set to true.
-  state.onClick = (ev: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement | HTMLElement>) => {
+  state.root.onClick = (ev: React.MouseEvent<HTMLAnchorElement> & React.MouseEvent<HTMLButtonElement>) => {
```

Comment:
(nit) Slightly more compact:
```suggestion
  state.root.onClick = (ev: React.MouseEvent<HTMLAnchorElement & HTMLButtonElement>) => {
```